synapse/module_api/__init__.py

```diff
@@ -889,6 +889,40 @@ async def update_room_membership(
 
         return event
 
+    async def update_m_direct(
```

Comment:
Sorry, I believe I have been misunderstood here.

What I was trying to say is that `update_m_direct` should live in the `synapse-auto-accept-invite` module.

Instead, I would suggest adding `add_account_data_for_user` (which is just a thin wrapper around `add_account_data_for_user`) and `get_account_data_for_user` (a thin wrapper again) to the Module API.

Reply:
Yep but as a first step I added just updating the list. I guess the logic for that is also better in the module, could we keep that also for a next step?

Reply:
The Module API is supposed to be a stable API. So I don't think we should add things to it if we're already planning to get rid of them soon, as it would mean we would need to maintain an unnecessary layer of backwards compatibility when we do the switch. So since we already have an idea of what the long-term solution should look like (and it's not a complex one) I think we should just do it.
As a nitpick, I think `add_account_data_for_user` should instead be named `set_account_data_for_user` - it makes it clearer that the method can be used for both creating new account data and updating existing one (sadly this makes it a bit inconsistent with the method from the account data handler, but I think it's best to make things clearer to module developers).

Reply:
> As a nitpick, I think add_account_data_for_user should instead be named set_account_data_for_user

No objections here with that; it does reflect that it will overwrite account data so sounds right.